### PR TITLE
ABI 1.1 support

### PR DESCRIPTION
**Pull request by tbfleming**
ABI 1.1 was introduced in https://github.com/EOSIO/eos/releases/tag/v1.3.0; it supports these new features:
* Variants: These are values which support multiple types. See https://github.com/EOSIO/eos/issues/5599 for details.
* Binary extensions: These support extending existing rows and actions with new fields. See https://github.com/EOSIO/eos/issues/5600 for details.

### CODE REVIEW

File: src/eosjs2-serialize.ts [hunk 1]
```diff
@@ -559,74 +573,122 @@ function deserializeUnknown(buffer: SerialBuffer): SerialBuffer {
     throw new Error("Don't know how to deserialize " + this.name);
 }
 
-function serializeStruct(buffer: SerialBuffer, data: any) {
+function serializeStruct(this: Type, buffer: SerialBuffer, data: any,
```
Comment (jeffreyssmith2nd) [inline src/eosjs2-serialize.ts]: Why the need to add `this` to a lot of these functions?

Comment (tbfleming) [reply src/eosjs2-serialize.ts]: It silently defaulted to `any`, defeating some type checking

Comment (tbfleming) [reply src/eosjs2-serialize.ts]: Oddly, the no-implicit-any check didn't catch it

Comment (jeffreyssmith2nd) [reply src/eosjs2-serialize.ts]: Interesting, I would not have expected this as the solution to that problem. Reminds me of Python's `self`.

Comment (tbfleming) [reply src/eosjs2-serialize.ts]: The weird part is it's still a hidden argument

**Comment by tbfleming** [comment]
Ready for review

File: src/eosjs2-api.test.ts [hunk 2]
```diff
@@ -169,10 +169,11 @@ describe("eosjs2-api", () => {
...
 
-    it("serializeTransaction converts tx to binary", () => {
-        const tx = api.serializeTransaction(transaction);
-        expect([...tx]).toEqual(serializedTx);
-    });
+    // This test misuses this function
+    // it("serializeTransaction converts tx to binary", () => {
+    //     const tx = api.serializeTransaction(transaction);
+    //     expect([...tx]).toEqual(serializedTx);
+    // });
```
Comment (brandonfancher) [inline src/eosjs2-api.test.ts]: See #66. This is commented out so that it can be fixed later.

File: src/eosjs2-serialize.ts (hunk 1 above)
Comment (brandonfancher) [reply src/eosjs2-serialize.ts]: https://www.typescriptlang.org/docs/handbook/functions.html#this-parameters

It's a neat TS feature.

**Pull request merged**